crates/sui-core/src/safe_client.rs

```diff
@@ -260,6 +301,51 @@ impl<C, S: SignatureVerifier> SafeClient<C, S> {
         }
     }
 
+    fn check_transaction_info_plain(
+        &self,
+        digest: &TransactionDigest,
+        transaction: VerifiedTransaction,
+        status: TransactionStatus,
+    ) -> SuiResult<PlainTransactionInfoResponse> {
+        fp_ensure!(
+            digest == transaction.digest(),
+            SuiError::ByzantineAuthoritySuspicion {
+                authority: self.address,
+                reason: "Signed transaction digest does not match with expected digest".to_string()
+            }
+        );
+        match status {
+            TransactionStatus::Signed(signed) => match self.get_committee(&signed.epoch) {
+                Ok(_) => Ok(PlainTransactionInfoResponse::Signed(
+                    SignedTransaction::new_from_data_and_sig(transaction.into_message(), signed),
+                )),
+                Err(e) => Err(e),
+            },
+            TransactionStatus::Executed(cert_opt, effects, events) => {
+                let signed_effects = self.check_signed_effects_plain(digest, effects, None)?;
+                match cert_opt {
+                    Some(cert) => {
+                        let committee = self.get_committee(&cert.epoch)?;
+                        Ok(PlainTransactionInfoResponse::ExecutedWithCert(
+                            CertifiedTransaction::new_from_data_and_sig(
+                                transaction.into_message(),
+                                cert,
+                            )
+                            .verify(&committee)?,
```

Comment:
what does this `verify` actually do here?

Reply:
todo: replace verified envelope with plain env

Reply:
discussed offline, this is still necessary